src/AutoRest.CSharp/Common/Generation/Writers/ModelFactoryWriter.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Common.Output.Builders;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models.Types;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal static class ModelFactoryWriter
+    {
+        public static IReadOnlyCollection<SchemaObjectType> ObjectTypesThatRequireFactory(IEnumerable<TypeProvider> models)
+            => models.OfType<SchemaObjectType>().Where(SchemaObjectTypeRequiresFactory).ToArray();
+
+        public static string WriteModelFactory(CodeWriter writer, BuildContext context, IEnumerable<SchemaObjectType> objectTypes)
+        {
+            var clientPrefix = ClientBuilder.GetClientPrefix(context.DefaultLibraryName, context);
+            var modelFactoryName = $"{clientPrefix}ModelFactory";
+            var @namespace = context.DefaultNamespace;
+
+            writer.UseNamespace("System.Linq");
```

Comment:
You shouldn't need this. When a type is used in `writer.Append` it keeps track of the namespaces.

Reply:
Removed